src/index/ivf_raft/ivf_raft.cuh

```diff
@@ -37,6 +37,70 @@
 
 namespace knowhere {
 
+namespace raft_res_pool {
+
+struct context {
+    context()
+        : resources_(
+              []() {
+                  return new rmm::cuda_stream();  // Avoid program exit datart
```

Comment:
None of these arguments are actually needed. By default, the `device_resouces` instance will use the `per_thread_default_cuda_stream` and the `workspace_resources` (the last argument to the constructor here) will automatically use `rmm::mr::get_current_device_resource()`